barricade/src/main/java/com/mutualmobile/barricade/Barricade.java

```diff
@@ -148,13 +139,40 @@ public long getDelay() {
     return delay;
   }
 
-  public void setDelay(long delay) {
-    this.delay = delay;
-  }
 
   @NonNull private String getResponseFromFile(String endpoint, String variant) {
     String fileName =
         ROOT_DIRECTORY + File.separator + endpoint + File.separator + variant + ".json";
     return fileManager.getContentsOfFileAsString(fileName);
   }
+
+  /**
+   * Methods for changing response and delay programmatically for testing
+   */
+
+  public Barricade enable() {
+    Logger.getLogger(TAG).info("Barricade enabled.");
+    this.disabled = false;
+    return this;
+  }
+
+  public Barricade disable() {
+    Logger.getLogger(TAG).info("Barricade disabled.");
+    this.disabled = true;
+    return this;
+  }
+
+  public Barricade setDelay(long delay) {
+    this.delay = delay;
+    return this;
+  }
+
+  public Barricade withResponse(String endPoint, int defaultIndex) {
```

Comment:
The default Index is the response index of the endpoint. Since using this method also changes the default index for that endpoint, hence the param name is 'default index'